pandas/core/internals/managers.py

```diff
@@ -956,7 +956,15 @@ def fast_xs(self, loc: int) -> ArrayLike:
             # Such assignment may incorrectly coerce NaT to None
             # result[blk.mgr_locs] = blk._slice((slice(None), loc))
             for i, rl in enumerate(blk.mgr_locs):
-                result[rl] = blk.iget((i, loc))
+                out = blk.iget((i, loc))
+                if is_dtype_equal(blk.dtype, dtype) and dtype == "m8[ns]":
+                    # FIXME: kludge for NaT -> tdnat
```

Comment:
i know you mention, but this is *very* kludgy.

Reply:
yah, im still looking for an alternative here

Reply:
de-kludged.  i need to double check, but think this may solve a separate silent bug.  if so ill make a separate PR for that